src/obj-ignore.c

```diff
@@ -322,9 +325,9 @@ int add_autoinscription(int16_t kind, const char *inscription, bool aware)
 	if (!inscription)
 		return remove_autoinscription(kind);
 	if (aware)
-		k->note_aware = quark_add(inscription);
+		k->note_aware = string_make(inscription);
 	else
-		k->note_unaware = quark_add(inscription);
+		k->note_unaware = string_make(inscription);
```

Comment:
Add `string_free(k->note_aware);` before the note_aware assignment and `string_free(k->note_unaware);`  before the note_unaware assignment to avoid memory leaks if the new auto inscription replaces an existing one.  The call to add_autoinscription() from ui-knowledge.c can't leak because it already calls remove_autoinscription(), but the call from ui-prefs.c could trigger a leak.